zone/doors.cpp

```diff
@@ -543,7 +543,7 @@ void Doors::HandleClick(Client *sender, uint8 trigger)
 		bool has_key_required = (required_key_item && required_key_item == player_key);
 
 		if (sender->GetGM() && has_key_required) {
-			has_key_required = false;
+			has_key_required = true;
```

Comment:
If `has_key_required` is already `true` based on the condition this is inside of, can’t we just remove this line altogether?

Reply:
Fixed it to predicate on `!has_key_required` so that the user is notified if their GM status changed the result of the interaction.